proposed/2023/NuGetAudit-without-nuget.org-as-packageSource.md

```diff
@@ -0,0 +1,190 @@
+   Would using nuget.org as a default audit source cause undue concern from customers who don't want to use nuget.org as a package source, and the new tooling version starts making network requests to nuget.org?
+
+   If we want to make nuget.org a default/automatic audit source, how?
+   NuGet has tried before to use a tracking file to automatically add nuget's v3 URL as a package source to user-profile nuget.config files that don't have any package sources.
+   However, this caused a security issue when customers wanted to remove nuget.org as a default package source, and this code incorrectly re-added it once (the customer could successfully remove nuget.org a second time).
+   When we removed this auto-update to mitigate the security concern, it exposed a bug where multiple other apps created user-profile nuget.config files without any package sources, and many customers assumed it was a bug with NuGet.
+
+   Should we hard-code nuget.org's default URL in NuGet.Configuration, and require customers to use `<clear/>` if they don't want any audit source?
+
+   Should we just add it as a default in new user-profile nuget.config files, and just accept that existing developers won't get it by default?
+
+2. Stop using package source as audit source?
+
+   On one hand, using package sources as a defacto audit source makes it easier for customers to use the feature when they knowingly or unknowingly use a package source with a VDB.
+   On the other hand, overloading `<packageSources>` with multiple meanings is confusing and customers probably won't understand why sometimes they get vulnerable package information and sometimes they don't.
+   If we add nuget.org as a default audit source, and don't use packageSources when auditSources has a value, then using packageSources to get a VDB when auditSources is empty will be particularly confusing/unintuitive to customers.
+
+   Another challenge is that if we want to change the behavior it' will be a breaking change, which is typically discouraged.
+   However, given that nuget.org is the only know source with a VDB, if we make nuget.org a default audit source, then it will effectively be a non-breaking change.
+
+3. Tooling
+
+   How important is it to add CLI commands to manage audit sources?
+   How important is it to add a GUI in Visual Studio to manage audit sources?
+
+4. Restore summary
+
+   When you restore at normal verbosity, NuGet outputs a list of package sources that were used for restore. Should we add audit sources as well?
+   If so, presumably we should only output it if at least 1 project in the restore has NuGetAudit enabled, as outputting audit sources when NuGetAudit is disabled for all projects might confuse customers.
+
+## Future Possibilities
+
+<!-- What future possibilities can you think of that this proposal would help with? -->
+
+### Allow auditSources to point to a filesystem directory
```

Comment:
I was thinking about offline scenarios as I read the spec, and was glad to see you mentioned it. However, I don't see this as a proper "Alternative", which I define as, "we decided this implementation isn't as good as the one proposed". 

Instead, this seems like a phase/iteration we should expect for this work. For the reasons you mentioned, I think supporting offline scenarios are important. The implementation doesn't need to be defined in this spec, but I'd like to see something concrete that it's a "real thing we will dig into in a (not too distant) future iteration" and not "an alternative we are casting out".

Reply:
This block you've commented on is under the "future possibilities" section, not "rationale and alternatives".

Reply:
You mentioned it in both sections. You even provided an in-document hyperlink to the Future Possibilities section from the [Rationale and alternatives](https://github.com/NuGet/Home/blob/zivkan-vulnerabilitySource-spec/proposed/2023/NuGetAudit-without-nuget.org-as-packageSource.md#rationale-and-alternatives) section. I feel it should be in "future" and not "alternatives".

https://github.com/NuGet/Home/blob/zivkan-vulnerabilitySource-spec/proposed/2023/NuGetAudit-without-nuget.org-as-packageSource.md#read-vulnerability-database-from-local-files

Reply:
I also have it under alternate designs, because I feel like implementing this first will give customers a workaround the "don't want to use nuget.org as a package source" scenario, and therefore `auditSources` would be unnecessary, and could save us from implementing the more complex design. I have it under both sections, because as an alternate design, it has UX issues, and I wanted to explain why it wasn't the chosen design.

Reply:
I watched the recording going through this spec, and reread this section. I think the "alternative" is that we're not shipping a dotnet tool that reads VDB files, and the "future" section is that we *will* support local audit sources? I hope your definition of the future section is such that it means an iteration or two from now, it'll be supported.

Reply:
Is there an issue for the local sources that can be linked here?
I would upvote it if I weren't on the team. Hopefully those in offline environments would see it and upvote it if interested in the feature.

Reply:
All the issues I create around NuGetAudit all have NuGetAudit in the title, making them [easy to search for](https://github.com/NuGet/Home/issues?q=is%3Aissue+NuGetAudit+is%3Aopen+).

I hadn't created one before, so I created one now:
* https://github.com/NuGet/Home/issues/12975

I'm not confident that many people will actually want this, so I marked it as in the icebox. We'll see if it gets enough upvotes before the bot closes it.